src/frame/headers.rs

```diff
@@ -309,17 +309,20 @@ impl fmt::Debug for Headers {
 
 // ===== util =====
 
-pub fn parse_u64(src: &[u8]) -> Result<u64, ()> {
+#[derive(Debug)]
+pub struct ParseU64Error;
+
+pub fn parse_u64(src: &[u8]) -> Result<u64, ParseU64Error> {
```

Comment:
im not totally sure about this change...can you point to the clippy lint?

Reply:
We can keep this for now but this is just an example of a clippy lint I don't totally agree with.

Reply:
I'd much prefer seeing `Err(ParseU64Error)` to `Err(())` in logs. Private code shouldn't be a free pass for odd api decisions, since it can always leak to the user one way or another (logs in this case).

Then again, since you've agreed to keep this I'm not really arguing for anything so I'll just get on with it and resolve the conversation.